### PR TITLE
Task-49004_49006_49602: Change the UI of the posting drawer for publishers

### PR DESCRIPTION
**Pull request by hassenelkamel-exo**
Prior to this change, we used a simple drawer to post news. After this enhancement, we implement a stepper which contains these steps: 
- Step 1: Post or not on the stream
- Step 2: Publish in a targeted section which is disabled with a feature flag until finishing the feature
- Step 3: Schedule (keep the old behavior)

### CODE REVIEW

File: webapp/src/main/webapp/skin/less/news.less [hunk 1]
```diff
@@ -3955,6 +3955,18 @@ input.ignore-vuetify-classes.datePickerText.flex-grow-0 {
...
+    max-width: ~"calc(100% - 24px)";
+  }
+  .scheduleDatePicker {
+    max-width: ~"calc(100% - 135px)";
+  }
+  .v-select .v-chip {
+    margin: 4px !important;
+  }
+  .chooseSpaceStream {
+    width: 95%;
```
Comment (azayati) [inline webapp/src/main/webapp/skin/less/news.less:3986]: to be tested on mobile

File: webapp/src/main/webapp/WEB-INF/conf/news/news-configuration.xml [hunk 2]
```diff
@@ -49,13 +49,13 @@
   </component>
 
   <component>
-    <key>ScheduleNewsFeatureProperties</key>
+    <key>TargetingNewsFeatureProperties</key>
```
Comment (azayati) [inline webapp/src/main/webapp/WEB-INF/conf/news/news-configuration.xml]:
```suggestion
    <key>NewsPublishTargetingFeatureProperties</key>
```

File: webapp/src/main/webapp/WEB-INF/conf/news/news-configuration.xml [hunk 3]
```diff
@@ -49,13 +49,13 @@
...
     <type>org.exoplatform.container.ExtendedPropertyConfigurator</type>
     <init-params>
       <properties-param>
-        <name>ScheduleNewsFeatureProperties</name>
-        <description>News schedule Feature enablement flag</description>
-        <property name="exo.feature.news.postLater.enabled" value="${exo.feature.news.postLater.enabled:false}" />
+        <name>TargetingNewsFeatureProperties</name>
```
Comment (azayati) [inline webapp/src/main/webapp/WEB-INF/conf/news/news-configuration.xml]:
```suggestion
        <name>NewsPublishTargetingFeatureProperties</name>
```

File: webapp/src/main/webapp/WEB-INF/conf/news/news-configuration.xml [hunk 4]
```diff
@@ -49,13 +49,13 @@
...
+        <description>News targeting Feature enablement flag</description>
```
Comment (azayati) [inline webapp/src/main/webapp/WEB-INF/conf/news/news-configuration.xml]:
```suggestion
        <description>News publish targeting Feature enablement flag</description>
```

File: webapp/src/main/webapp/WEB-INF/conf/news/news-configuration.xml [hunk 5]
```diff
@@ -49,13 +49,13 @@
...
+        <property name="exo.feature.news.targetingPublish.enabled" value="${exo.feature.news.targetingPublish.enabled:false}" />
```
Comment (azayati) [inline webapp/src/main/webapp/WEB-INF/conf/news/news-configuration.xml]:
```suggestion
        <property name="exo.feature.news.publishTargeting.enabled" value="${exo.feature.news.publishTargeting.enabled:false}" />
```

File: webapp/src/main/webapp/news-details/components/ExoNewsDetailsBody.vue [hunk 6]
```diff
@@ -75,7 +75,7 @@
                         :format="dateTimeFormat"
                         class="newsInformationValue newsUpdatedDate ml-1 me-1" />
                     </template>
-                    <div v-else-if="news.updatedDate" class="newsInformationValue newsUpdatedDate">{{ news.updatedDate }}</div>
+                    <div v-else-if="updatedDate" class="newsInformationValue newsUpdatedDate">{{ updatedDate }}</div>
```
Comment (azayati) [inline webapp/src/main/webapp/news-details/components/ExoNewsDetailsBody.vue:78]: why this modification ?

File: webapp/src/main/webapp/schedule-news-drawer/components/ExoScheduleNewsDrawer.vue [hunk 7]
```diff
@@ -19,68 +20,159 @@
...
+        <v-stepper
+          v-model="stepper"
+          vertical
+          flat
+          class="ma-0 py-0 me-4">
+          <v-stepper-step
+            :complete="stepper > 1"
+            step="1"
+            class="ma-0">
+            {{ $t('news.composer.stepper.postStream') }}
```
Comment (azayati) [inline webapp/src/main/webapp/schedule-news-drawer/components/ExoScheduleNewsDrawer.vue]:
```suggestion
            {{ $t('news.composer.stepper.postOnStream.title') }}
```

File: webapp/src/main/webapp/schedule-news-drawer/components/ExoScheduleNewsDrawer.vue [hunk 8]
```diff
@@ -19,68 +20,159 @@
...
+          </v-stepper-step>
+          <v-stepper-content step="1" class="ps-4 pe-6 my-0">
+            <div class="d-flex flex-column pt-2">
+              <div class="d-flex flex-row grey--text ms-2 chooseSpaceStream">{{ $t('news.composer.stepper.chooseStream') }}</div>
```
Comment (azayati) [inline webapp/src/main/webapp/schedule-news-drawer/components/ExoScheduleNewsDrawer.vue]:
```suggestion
              <div class="d-flex flex-row grey--text ms-2 postOnStreamOption">{{ $t('news.composer.stepper.postOnStream.option') }}</div>
```

File: webapp/src/main/webapp/schedule-news-drawer/components/ExoScheduleNewsDrawer.vue [hunk 9]
```diff
@@ -19,68 +20,159 @@
...
+              <div class="d-flex flex-row">
+                <v-switch
+                  inset
+                  dense
+                  value
+                  input-value="true"
+                  disabled
+                  class="my-0 ms-3" />
+                <label class="my-auto">
+                  {{ $t('news.composer.stepper.postActivityStream') }}
```
Comment (azayati) [inline webapp/src/main/webapp/schedule-news-drawer/components/ExoScheduleNewsDrawer.vue]:
```suggestion
                  {{ $t('news.composer.stepper.postOnStream.description') }}
```

File: webapp/src/main/webapp/schedule-news-drawer/components/ExoScheduleNewsDrawer.vue [hunk 10]
```diff
@@ -19,68 +20,159 @@
...
+                  </v-icon>
+                </v-btn>
+              </v-card-actions>
+            </div>
+          </v-stepper-content>
+          <v-stepper-step
+            :complete="stepper > 2"
+            step="2"
+            class="ma-0">
+            {{ $t('news.composer.stepper.publishSection') }}
```
Comment (azayati) [inline webapp/src/main/webapp/schedule-news-drawer/components/ExoScheduleNewsDrawer.vue]:
```suggestion
            {{ $t('news.composer.stepper.publishSection.title') }}
```

File: webapp/src/main/webapp/schedule-news-drawer/components/ExoScheduleNewsDrawer.vue [hunk 11]
```diff
@@ -19,68 +20,159 @@
...
+          </v-stepper-step>
+          <v-stepper-content step="2" class="ps-4 pe-6 my-0">
+            <div class="d-flex flex-column pt-2">
+              <div class="d-flex flex-row grey--text ms-2">{{ $t('news.composer.stepper.publishDedicatedSection') }}</div>
```
Comment (azayati) [inline webapp/src/main/webapp/schedule-news-drawer/components/ExoScheduleNewsDrawer.vue]:
```suggestion
              <div class="d-flex flex-row grey--text ms-2">{{ $t('news.composer.stepper.publishSection.description') }}</div>
```

File: webapp/src/main/webapp/schedule-news-drawer/components/ExoScheduleNewsDrawer.vue [hunk 12]
```diff
@@ -19,68 +20,159 @@
...
+                <v-switch
+                  v-model="publish"
+                  inset
+                  dense
+                  class="my-0 ms-3" />
+                <label class="my-auto">
+                  {{ $t('news.details.header.menu.publish') }}
+                </label>
+              </div>
+              <div v-if="allowTargetingPublish" class="d-flex flex-row grey--text ms-2">{{ $t('news.composer.stepper.selectedTarget') }}</div>
```
Comment (azayati) [inline webapp/src/main/webapp/schedule-news-drawer/components/ExoScheduleNewsDrawer.vue]:
```suggestion
              <div v-if="allowPublishTargeting" class="d-flex flex-row grey--text ms-2">{{ $t('news.composer.stepper.selectedTarget.description') }}</div>
```

File: webapp/src/main/webapp/schedule-news-drawer/components/ExoScheduleNewsDrawer.vue [hunk 13]
```diff
@@ -19,68 +20,159 @@
...
+              <div v-if="allowTargetingPublish" class="d-flex flex-row selectTarget ms-2">
+                <v-select
+                  id="selectTarget"
+                  ref="selectTarget"
+                  :items="items"
+                  :placeholder="$t('news.composer.stepper.chooseTarget')"
```
Comment (azayati) [inline webapp/src/main/webapp/schedule-news-drawer/components/ExoScheduleNewsDrawer.vue]:
```suggestion
                  :placeholder="$t('news.composer.stepper.chooseTargets.option)"
```

File: webapp/src/main/webapp/schedule-news-drawer/components/ExoScheduleNewsDrawer.vue [hunk 14]
```diff
@@ -19,68 +20,159 @@
...
+                  class="my-0 ms-3" />
+                <label class="my-auto">
+                  {{ $t('news.details.header.menu.publish') }}
+                </label>
+              </div>
+              <div v-if="allowTargetingPublish" class="d-flex flex-row grey--text ms-2">{{ $t('news.composer.stepper.selectedTarget') }}</div>
+              <div v-if="allowTargetingPublish" class="d-flex flex-row selectTarget ms-2">
+                <v-select
+                  id="selectTarget"
+                  ref="selectTarget"
```
Comment (azayati) [inline webapp/src/main/webapp/schedule-news-drawer/components/ExoScheduleNewsDrawer.vue]:
```suggestion
                  id="chooseTargets"
                  ref="chooseTargets"
```

**Review by azayati — CHANGES_REQUESTED** [review]
See review

File: webapp/src/main/webapp/skin/less/news.less (hunk 1 above)
Comment (hassenelkamel-exo) [reply webapp/src/main/webapp/skin/less/news.less:3986]: well tested in mobile view

File: webapp/src/main/webapp/news-details/components/ExoNewsDetailsBody.vue (hunk 6 above)
Comment (hassenelkamel-exo) [reply webapp/src/main/webapp/news-details/components/ExoNewsDetailsBody.vue:78]: some times when reload details we have null value so we use `updatedDate` computed value

File: webapp/src/main/webapp/schedule-news-drawer/components/ExoScheduleNewsDrawer.vue [hunk 15]
```diff
@@ -19,68 +20,159 @@
...
+          class="ma-0 py-0 me-4">
+          <v-stepper-step
+            :complete="stepper > 1"
+            step="1"
+            class="ma-0">
+            {{ $t('news.composer.stepper.postStream.title') }}
+          </v-stepper-step>
+          <v-stepper-content step="1" class="ps-4 pe-6 my-0">
+            <div class="d-flex flex-column pt-2">
+              <div class="d-flex flex-row grey--text ms-2 postOnStreamOption">{{ $t('news.composer.stepper.postOnStream.option') }}</div>
```
Comment (azayati) [inline webapp/src/main/webapp/schedule-news-drawer/components/ExoScheduleNewsDrawer.vue]:
```suggestion
              <div class="d-flex flex-row grey--text ms-2 postOnStreamOption">{{ $t('news.composer.stepper.postOnStream.description') }}</div>
```

File: webapp/src/main/webapp/schedule-news-drawer/components/ExoScheduleNewsDrawer.vue [hunk 16]
```diff
@@ -19,68 +20,159 @@
...
+              <div class="d-flex flex-row">
+                <v-switch
+                  inset
+                  dense
+                  value
+                  input-value="true"
+                  disabled
+                  class="my-0 ms-3" />
+                <label class="my-auto">
+                  {{ $t('news.composer.stepper.postOnStream.description') }}
```
Comment (azayati) [inline webapp/src/main/webapp/schedule-news-drawer/components/ExoScheduleNewsDrawer.vue]:
```suggestion
                  {{ $t('news.composer.stepper.postOnStream.option') }}
```

File: webapp/src/main/webapp/schedule-news-drawer/components/ExoScheduleNewsDrawer.vue [hunk 17]
```diff
@@ -19,68 +20,159 @@
...
+            <div class="d-flex flex-column pt-2">
+              <div class="d-flex flex-row grey--text ms-2">{{ $t('news.composer.stepper.publishSection.description') }}</div>
+              <div class="d-flex flex-row">
+                <v-switch
+                  v-model="publish"
+                  inset
+                  dense
+                  class="my-0 ms-3" />
+                <label class="my-auto">
+                  {{ $t('news.details.header.menu.publish') }}
```
Comment (azayati) [inline webapp/src/main/webapp/schedule-news-drawer/components/ExoScheduleNewsDrawer.vue]:
```suggestion
                  {{ $t('news.composer.publishSection.option') }}
```

File: webapp/src/main/webapp/schedule-news-drawer/components/ExoScheduleNewsDrawer.vue [hunk 18]
```diff
@@ -19,68 +20,159 @@
...
+                  </v-icon>
+                </v-btn>
+              </v-card-actions>
+            </div>
+          </v-stepper-content>
+          <v-stepper-step
+            :complete="stepper > 3"
+            step="3"
+            class="ma-0">
+            {{ $t('news.composer.schedule') }}
```
Comment (azayati) [inline webapp/src/main/webapp/schedule-news-drawer/components/ExoScheduleNewsDrawer.vue]:
```suggestion
            {{ $t(news.composer.stepper.schedule.title') }}
```

**Review by azayati — CHANGES_REQUESTED** [review]
see review

**Review by azayati — DISMISSED** [review]
LGTM after QA validation

**Review by azayati — APPROVED** [review]
LGTM

**Pull request merged**